server/handler/query.go

```diff
@@ -62,41 +66,60 @@ func Query(db service.SQLDB) RequestProcessFunc {
 				`Bad Request. Expected body: { "query": "SQL statement", "limit": 1234 }`)
 		}
 
-		query, limitSet := addLimit(queryRequest.Query, queryRequest.Limit)
-		rows, err := db.Query(query)
-		if err != nil {
-			return nil, dbError(err)
-		}
-		defer rows.Close()
+		respCh := make(chan *serializer.Response)
+		errCh := make(chan error)
 
-		columnNames, columnTypes, err := columnsInfo(rows)
-		if err != nil {
-			return nil, err
-		}
+		go func() {
```

Comment:
could you please explain why do we need to wrap it in goroutine?

Reply:
My intention was to wrap all the logic to be able to cancel not only the db query, but also the potentially costly processing on the results. But, reading it again, I realize it's no use because the code would need to listen for the context cancellation in more places.

I'll remove the goroutine and simplify to 
```
		rows, err := db.QueryContext(ctx, query)
		if err != nil {
			return nil, err
		}
```